### PR TITLE
Frontend react setup

### PR DESCRIPTION
**Pull request by LittenJV**
initial pull request for adding login page with nice css. Login can't persist just yet

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@material-ui/core": "^4.11.0",
+    "axios": "^0.20.0",
```
Comment (benedict-soh) [inline frontend/package.json:7]: pog

File: frontend/src/App.js [hunk 2]
```diff
@@ -3,14 +3,31 @@ import NavigationTopBar from './navigation/NavigationTopBar'
...
+        super(props)
+    }
+
+    componentWillMount() {
+        this.check_auth()
+    }
+
+    //check authentication
+    check_auth() {
+        console.log("checking")
```
Comment (benedict-soh) [inline frontend/src/App.js:20]: Nice check. 10/10

File: frontend/src/login/login.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import Button from "@material-ui/core/Button";
+import Grid from "@material-ui/core/Grid";
+import {login_request} from "../utils/API";
+import { Alert } from '@material-ui/lab';
+
+export default class loginPage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isLoading: false,
```
Comment (benedict-soh) [inline frontend/src/login/login.js:17]: Loading bar? PogChamp

File: frontend/src/App.js (hunk 2 above)
Comment (LittenJV) [reply frontend/src/App.js:20]: works 1000% of the time

**Review by benedict-soh — APPROVED** [review]
Looks good I guess

**Pull request merged**